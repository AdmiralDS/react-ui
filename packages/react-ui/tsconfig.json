{
  "compilerOptions": {
    "target": "ES2019",
    "lib": ["DOM", "DOM.Iterable", "ES2019"],
    "types": ["@types/jest", "@testing-library/dom", "@testing-library/react", "@types/testing-library__jest-dom"],
    "allowJs": true,
    "skipLibCheck": false,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "module": "commonjs",
    "moduleResolution": "Node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "declaration": true,
    "declarationMap": true,
    "emitDeclarationOnly": true,
    "declarationDir": "./dist",
    "jsx": "react",
    "baseUrl": "./src",
    "paths": {
      "#src/*": ["./*"]
    },
    "plugins": [{ "transform": "typescript-transform-paths", "afterDeclarations": true }]
  },
  "include": ["./src"],
  "exclude": ["./scripts"]
}
