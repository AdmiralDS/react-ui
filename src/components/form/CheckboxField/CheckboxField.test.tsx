import { LIGHT_THEME, CheckboxField } from '@admiral-ds/react-ui';
import { ThemeProvider } from 'styled-components';
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('CheckboxField', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const CheckboxFieldRequiredProps = {
    onChange: jest.fn(),
    checked: false,
    id: 'checkboxField',

    'data-testid': 'element',
  };

  it('should render component with required properties', () => {
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <CheckboxField {...CheckboxFieldRequiredProps}>text</CheckboxField>
      </ThemeProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with disabled state', () => {
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <CheckboxField {...CheckboxFieldRequiredProps} disabled>
          text
        </CheckboxField>
      </ThemeProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with small dimension', () => {
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <CheckboxField {...CheckboxFieldRequiredProps} dimension="s">
          text
        </CheckboxField>
      </ThemeProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onChange when user clicks on component', () => {
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <CheckboxField {...CheckboxFieldRequiredProps}>text</CheckboxField>
      </ThemeProvider>,
    );
    fireEvent.click(wrapper.getByTestId('element'));
    const { onChange } = CheckboxFieldRequiredProps;
    expect(onChange).toBeCalledTimes(1);
  });

  it('should focus input if user press Tab key', async () => {
    const user = userEvent.setup();
    render(
      <ThemeProvider theme={LIGHT_THEME}>
        <CheckboxField {...CheckboxFieldRequiredProps}>text</CheckboxField>
      </ThemeProvider>,
    );
    const [radio] = screen.getAllByTestId('element');
    await user.tab();
    expect(radio).toHaveFocus();
  });

  it('should call onChange if input is focused and user press Space key', async () => {
    const user = userEvent.setup();
    render(
      <ThemeProvider theme={LIGHT_THEME}>
        <CheckboxField {...CheckboxFieldRequiredProps}>text</CheckboxField>
      </ThemeProvider>,
    );
    const [CheckboxFieldHTMLElement] = screen.getAllByTestId('element');
    await user.tab();
    await user.type(CheckboxFieldHTMLElement, '{Space}');
    const { onChange } = CheckboxFieldRequiredProps;
    expect(onChange).toBeCalledTimes(1);
  });
});
