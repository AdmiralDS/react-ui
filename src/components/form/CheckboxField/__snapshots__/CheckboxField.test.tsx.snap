// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxField should render component with disabled state 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c8 {
  pointer-events: none;
}

.c8 *[fill^='#'] {
  fill: #FFFFFF;
}

.c3 {
  position: relative;
  box-sizing: border-box;
  overflow: visible;
  width: 20px;
  height: 20px;
}

fieldset[data-dimension='s'] .c2 {
  width: 16px;
}

fieldset[data-dimension='s'] .c2 {
  height: 16px;
}

.c7 {
  pointer-events: none;
  display: inline-block;
  position: absolute;
  margin: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 4px;
  outline: 0;
  -webkit-transition: background 0.1s ease-in;
  transition: background 0.1s ease-in;
  line-height: initial;
  background-color: #FFFFFF;
  border: 1px solid #626F84;
}

.c5 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: calc(100% + 2px);
  height: calc(100% + 2px);
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  box-sizing: border-box;
  cursor: pointer;
  border-radius: 5px;
  margin: 0;
  padding: 0;
}

.c5::-ms-check {
  display: none;
}

.c5:disabled {
  pointer-events: none;
}

.c5:checked + .c6 {
  background-color: #0D69F2;
  border: none;
}

.c5:not(:checked) + .c6 > * {
  display: none;
}

.c5:not(:checked):focus + .c6,
.c5:not(:disabled):not(:checked):hover + .c6 {
  background-color: #F1F2F4;
  border: 1px solid #0A54C2;
}

.c5:checked:focus + .c6,
.c5:not(:disabled):checked:hover + .c6 {
  background-color: #0A54C2;
  border: none;
}

.c5:not(:checked):disabled + .c6 {
  background-color: #FFFFFF;
  border: 1px solid #C4CAD4;
}

.c5:disabled + .c6 {
  background-color: #C4CAD4;
  border: none;
}

.c5:focus-visible {
  outline-offset: 2px;
  outline: #0A54C2 solid 2px;
}

.c4 {
  position: absolute;
  top: 2px;
  left: 2px;
}

.c1 {
  display: inline-block;
  position: relative;
  padding-top: 2px;
  padding-bottom: 2px;
  padding-left: 32px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  color: #A7B0BE;
  cursor: default;
}

fieldset[data-dimension='s'] .c0 {
  padding-left: 28px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 16px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

fieldset:disabled .c0 {
  color: #A7B0BE;
  cursor: default;
}

<body>
    <div>
      <label
        class="c0 c1"
        disabled=""
      >
        <div
          class="c2 c3 c4"
        >
          <input
            class="c5"
            data-testid="element"
            disabled=""
            type="checkbox"
          />
          <div
            class="c6 c7"
          >
            <div
              aria-hidden="true"
              class="c8"
              focusable="false"
            />
          </div>
        </div>
        text
      </label>
    </div>
  </body>,
  "container": .c8 {
  pointer-events: none;
}

.c8 *[fill^='#'] {
  fill: #FFFFFF;
}

.c3 {
  position: relative;
  box-sizing: border-box;
  overflow: visible;
  width: 20px;
  height: 20px;
}

fieldset[data-dimension='s'] .c2 {
  width: 16px;
}

fieldset[data-dimension='s'] .c2 {
  height: 16px;
}

.c7 {
  pointer-events: none;
  display: inline-block;
  position: absolute;
  margin: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 4px;
  outline: 0;
  -webkit-transition: background 0.1s ease-in;
  transition: background 0.1s ease-in;
  line-height: initial;
  background-color: #FFFFFF;
  border: 1px solid #626F84;
}

.c5 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: calc(100% + 2px);
  height: calc(100% + 2px);
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  box-sizing: border-box;
  cursor: pointer;
  border-radius: 5px;
  margin: 0;
  padding: 0;
}

.c5::-ms-check {
  display: none;
}

.c5:disabled {
  pointer-events: none;
}

.c5:checked + .c6 {
  background-color: #0D69F2;
  border: none;
}

.c5:not(:checked) + .c6 > * {
  display: none;
}

.c5:not(:checked):focus + .c6,
.c5:not(:disabled):not(:checked):hover + .c6 {
  background-color: #F1F2F4;
  border: 1px solid #0A54C2;
}

.c5:checked:focus + .c6,
.c5:not(:disabled):checked:hover + .c6 {
  background-color: #0A54C2;
  border: none;
}

.c5:not(:checked):disabled + .c6 {
  background-color: #FFFFFF;
  border: 1px solid #C4CAD4;
}

.c5:disabled + .c6 {
  background-color: #C4CAD4;
  border: none;
}

.c5:focus-visible {
  outline-offset: 2px;
  outline: #0A54C2 solid 2px;
}

.c4 {
  position: absolute;
  top: 2px;
  left: 2px;
}

.c1 {
  display: inline-block;
  position: relative;
  padding-top: 2px;
  padding-bottom: 2px;
  padding-left: 32px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  color: #A7B0BE;
  cursor: default;
}

fieldset[data-dimension='s'] .c0 {
  padding-left: 28px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 16px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

fieldset:disabled .c0 {
  color: #A7B0BE;
  cursor: default;
}

<div>
    <label
      class="c0 c1"
      disabled=""
    >
      <div
        class="c2 c3 c4"
      >
        <input
          class="c5"
          data-testid="element"
          disabled=""
          type="checkbox"
        />
        <div
          class="c6 c7"
        >
          <div
            aria-hidden="true"
            class="c8"
            focusable="false"
          />
        </div>
      </div>
      text
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CheckboxField should render component with required properties 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c8 {
  pointer-events: none;
}

.c8 *[fill^='#'] {
  fill: #FFFFFF;
}

.c3 {
  position: relative;
  box-sizing: border-box;
  overflow: visible;
  width: 20px;
  height: 20px;
}

fieldset[data-dimension='s'] .c2 {
  width: 16px;
}

fieldset[data-dimension='s'] .c2 {
  height: 16px;
}

.c7 {
  pointer-events: none;
  display: inline-block;
  position: absolute;
  margin: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 4px;
  outline: 0;
  -webkit-transition: background 0.1s ease-in;
  transition: background 0.1s ease-in;
  line-height: initial;
  background-color: #FFFFFF;
  border: 1px solid #626F84;
}

.c5 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: calc(100% + 2px);
  height: calc(100% + 2px);
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  box-sizing: border-box;
  cursor: pointer;
  border-radius: 5px;
  margin: 0;
  padding: 0;
}

.c5::-ms-check {
  display: none;
}

.c5:disabled {
  pointer-events: none;
}

.c5:checked + .c6 {
  background-color: #0D69F2;
  border: none;
}

.c5:not(:checked) + .c6 > * {
  display: none;
}

.c5:not(:checked):focus + .c6,
.c5:not(:disabled):not(:checked):hover + .c6 {
  background-color: #F1F2F4;
  border: 1px solid #0A54C2;
}

.c5:checked:focus + .c6,
.c5:not(:disabled):checked:hover + .c6 {
  background-color: #0A54C2;
  border: none;
}

.c5:not(:checked):disabled + .c6 {
  background-color: #FFFFFF;
  border: 1px solid #C4CAD4;
}

.c5:disabled + .c6 {
  background-color: #C4CAD4;
  border: none;
}

.c5:focus-visible {
  outline-offset: 2px;
  outline: #0A54C2 solid 2px;
}

.c4 {
  position: absolute;
  top: 2px;
  left: 2px;
}

.c1 {
  display: inline-block;
  position: relative;
  padding-top: 2px;
  padding-bottom: 2px;
  padding-left: 32px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  color: #2B313B;
  cursor: pointer;
}

fieldset[data-dimension='s'] .c0 {
  padding-left: 28px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 16px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

fieldset:disabled .c0 {
  color: #A7B0BE;
  cursor: default;
}

<body>
    <div>
      <label
        class="c0 c1"
      >
        <div
          class="c2 c3 c4"
        >
          <input
            class="c5"
            data-testid="element"
            type="checkbox"
          />
          <div
            class="c6 c7"
          >
            <div
              aria-hidden="true"
              class="c8"
              focusable="false"
            />
          </div>
        </div>
        text
      </label>
    </div>
  </body>,
  "container": .c8 {
  pointer-events: none;
}

.c8 *[fill^='#'] {
  fill: #FFFFFF;
}

.c3 {
  position: relative;
  box-sizing: border-box;
  overflow: visible;
  width: 20px;
  height: 20px;
}

fieldset[data-dimension='s'] .c2 {
  width: 16px;
}

fieldset[data-dimension='s'] .c2 {
  height: 16px;
}

.c7 {
  pointer-events: none;
  display: inline-block;
  position: absolute;
  margin: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 4px;
  outline: 0;
  -webkit-transition: background 0.1s ease-in;
  transition: background 0.1s ease-in;
  line-height: initial;
  background-color: #FFFFFF;
  border: 1px solid #626F84;
}

.c5 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: calc(100% + 2px);
  height: calc(100% + 2px);
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  box-sizing: border-box;
  cursor: pointer;
  border-radius: 5px;
  margin: 0;
  padding: 0;
}

.c5::-ms-check {
  display: none;
}

.c5:disabled {
  pointer-events: none;
}

.c5:checked + .c6 {
  background-color: #0D69F2;
  border: none;
}

.c5:not(:checked) + .c6 > * {
  display: none;
}

.c5:not(:checked):focus + .c6,
.c5:not(:disabled):not(:checked):hover + .c6 {
  background-color: #F1F2F4;
  border: 1px solid #0A54C2;
}

.c5:checked:focus + .c6,
.c5:not(:disabled):checked:hover + .c6 {
  background-color: #0A54C2;
  border: none;
}

.c5:not(:checked):disabled + .c6 {
  background-color: #FFFFFF;
  border: 1px solid #C4CAD4;
}

.c5:disabled + .c6 {
  background-color: #C4CAD4;
  border: none;
}

.c5:focus-visible {
  outline-offset: 2px;
  outline: #0A54C2 solid 2px;
}

.c4 {
  position: absolute;
  top: 2px;
  left: 2px;
}

.c1 {
  display: inline-block;
  position: relative;
  padding-top: 2px;
  padding-bottom: 2px;
  padding-left: 32px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  color: #2B313B;
  cursor: pointer;
}

fieldset[data-dimension='s'] .c0 {
  padding-left: 28px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 16px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

fieldset:disabled .c0 {
  color: #A7B0BE;
  cursor: default;
}

<div>
    <label
      class="c0 c1"
    >
      <div
        class="c2 c3 c4"
      >
        <input
          class="c5"
          data-testid="element"
          type="checkbox"
        />
        <div
          class="c6 c7"
        >
          <div
            aria-hidden="true"
            class="c8"
            focusable="false"
          />
        </div>
      </div>
      text
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CheckboxField should render component with small dimension 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c8 {
  pointer-events: none;
}

.c8 *[fill^='#'] {
  fill: #FFFFFF;
}

.c3 {
  position: relative;
  box-sizing: border-box;
  overflow: visible;
  width: 16px;
  height: 16px;
}

fieldset[data-dimension='s'] .c2 {
  width: 16px;
}

fieldset[data-dimension='s'] .c2 {
  height: 16px;
}

.c7 {
  pointer-events: none;
  display: inline-block;
  position: absolute;
  margin: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 4px;
  outline: 0;
  -webkit-transition: background 0.1s ease-in;
  transition: background 0.1s ease-in;
  line-height: initial;
  background-color: #FFFFFF;
  border: 1px solid #626F84;
}

.c5 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: calc(100% + 2px);
  height: calc(100% + 2px);
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  box-sizing: border-box;
  cursor: pointer;
  border-radius: 5px;
  margin: 0;
  padding: 0;
}

.c5::-ms-check {
  display: none;
}

.c5:disabled {
  pointer-events: none;
}

.c5:checked + .c6 {
  background-color: #0D69F2;
  border: none;
}

.c5:not(:checked) + .c6 > * {
  display: none;
}

.c5:not(:checked):focus + .c6,
.c5:not(:disabled):not(:checked):hover + .c6 {
  background-color: #F1F2F4;
  border: 1px solid #0A54C2;
}

.c5:checked:focus + .c6,
.c5:not(:disabled):checked:hover + .c6 {
  background-color: #0A54C2;
  border: none;
}

.c5:not(:checked):disabled + .c6 {
  background-color: #FFFFFF;
  border: 1px solid #C4CAD4;
}

.c5:disabled + .c6 {
  background-color: #C4CAD4;
  border: none;
}

.c5:focus-visible {
  outline-offset: 2px;
  outline: #0A54C2 solid 2px;
}

.c4 {
  position: absolute;
  top: 2px;
  left: 2px;
}

.c1 {
  display: inline-block;
  position: relative;
  padding-top: 2px;
  padding-bottom: 2px;
  padding-left: 28px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 16px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  color: #2B313B;
  cursor: pointer;
}

fieldset[data-dimension='s'] .c0 {
  padding-left: 28px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 16px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

fieldset:disabled .c0 {
  color: #A7B0BE;
  cursor: default;
}

<body>
    <div>
      <label
        class="c0 c1"
      >
        <div
          class="c2 c3 c4"
        >
          <input
            class="c5"
            data-testid="element"
            type="checkbox"
          />
          <div
            class="c6 c7"
          >
            <div
              aria-hidden="true"
              class="c8"
              focusable="false"
            />
          </div>
        </div>
        text
      </label>
    </div>
  </body>,
  "container": .c8 {
  pointer-events: none;
}

.c8 *[fill^='#'] {
  fill: #FFFFFF;
}

.c3 {
  position: relative;
  box-sizing: border-box;
  overflow: visible;
  width: 16px;
  height: 16px;
}

fieldset[data-dimension='s'] .c2 {
  width: 16px;
}

fieldset[data-dimension='s'] .c2 {
  height: 16px;
}

.c7 {
  pointer-events: none;
  display: inline-block;
  position: absolute;
  margin: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 4px;
  outline: 0;
  -webkit-transition: background 0.1s ease-in;
  transition: background 0.1s ease-in;
  line-height: initial;
  background-color: #FFFFFF;
  border: 1px solid #626F84;
}

.c5 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: calc(100% + 2px);
  height: calc(100% + 2px);
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  box-sizing: border-box;
  cursor: pointer;
  border-radius: 5px;
  margin: 0;
  padding: 0;
}

.c5::-ms-check {
  display: none;
}

.c5:disabled {
  pointer-events: none;
}

.c5:checked + .c6 {
  background-color: #0D69F2;
  border: none;
}

.c5:not(:checked) + .c6 > * {
  display: none;
}

.c5:not(:checked):focus + .c6,
.c5:not(:disabled):not(:checked):hover + .c6 {
  background-color: #F1F2F4;
  border: 1px solid #0A54C2;
}

.c5:checked:focus + .c6,
.c5:not(:disabled):checked:hover + .c6 {
  background-color: #0A54C2;
  border: none;
}

.c5:not(:checked):disabled + .c6 {
  background-color: #FFFFFF;
  border: 1px solid #C4CAD4;
}

.c5:disabled + .c6 {
  background-color: #C4CAD4;
  border: none;
}

.c5:focus-visible {
  outline-offset: 2px;
  outline: #0A54C2 solid 2px;
}

.c4 {
  position: absolute;
  top: 2px;
  left: 2px;
}

.c1 {
  display: inline-block;
  position: relative;
  padding-top: 2px;
  padding-bottom: 2px;
  padding-left: 28px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 16px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  color: #2B313B;
  cursor: pointer;
}

fieldset[data-dimension='s'] .c0 {
  padding-left: 28px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 16px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

fieldset:disabled .c0 {
  color: #A7B0BE;
  cursor: default;
}

<div>
    <label
      class="c0 c1"
    >
      <div
        class="c2 c3 c4"
      >
        <input
          class="c5"
          data-testid="element"
          type="checkbox"
        />
        <div
          class="c6 c7"
        >
          <div
            aria-hidden="true"
            class="c8"
            focusable="false"
          />
        </div>
      </div>
      text
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
